### ISSUE TITLE
Connecting Viva Wallet to WooCommerce Critical Error

### ISSUE THREAD

**Issue by Si17br18** [VivaPayments/API#1777]
Hi Im trying to connect VW to WC. I followed the link they provided and downloaded the zip file. Followed all the steps correctly except for uploading their plugin to WP. When I do this it fails to upload the PHP file. To resolve this I searched for the plugin through WP and installed it. everything seems connected fine and when I go to checkout I get the following error:

Fatal error: Uncaught Exception: Unable to reach Viva Payments (Not Found) in /home/customer/www/hempharmony.co/public_html/wp-content/plugins/vivawallet-woocommerce-gateway/wc-vivawallet.php:310 Stack trace: #0 /home/customer/www/hempharmony.co/public_html/wp-content/plugins/vivawallet-woocommerce-gateway/wc-vivawallet.php(428): WC_VIVAWALLET->generate_form(582) #1 /home/customer/www/hempharmony.co/public_html/wp-includes/class-wp-hook.php(287): WC_VIVAWALLET->receipt_page(582) #2 /home/customer/www/hempharmony.co/public_html/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters('', Array) #3 /home/customer/www/hempharmony.co/public_html/wp-includes/plugin.php(478): WP_Hook->do_action(Array) #4 /home/customer/www/hempharmony.co/public_html/wp-content/plugins/woocommerce/templates/checkout/order-receipt.php(44): do_action('woocommerce_rec...', 582) #5 /home/customer/www/hempharmony.co/public_html/wp-content/plugins/woocommerce/includes/wc-core-functions.php(344): include('/home/customer/...') #6 /home/customer/www/he in /home/customer/www/hempharmony.co/public_html/wp-content/plugins/vivawallet-woocommerce-gateway/wc-vivawallet.php on line 310

please feel free to visit the site and see for yourself 

www.hempharmony.co

Kind regards
Simon

**Comment by matthewrgourd** [comment]
HI @Si17br18, sorry to hear this. We now have two versions of our plugin for WC available, 3.6.2 (redirect checkout) and 1.0.1 (native). The second one is the most up-to-date so you should install that one. The relevant links are below:

**VivaWallet WooCommerce Gateway (v3.6.2)**
[Download page](https://en-gb.wordpress.org/plugins/vivawallet-woocommerce-gateway/)
[Documentation](https://developer.vivawallet.com/e-commerce-plugins/woocommerce/)

**Viva Wallet for WooCommerce (v1.0.1)**
[Download page](https://woocommerce.com/products/viva-wallet-for-woocommerce)
[Documentation](https://docs.woocommerce.com/document/viva-wallet-for-woocommerce)

Which one are you referring to? Thanks!

**Comment by matthewrgourd** [comment]
By the way, you should upload the entire ZIP rather than individual files to WP.

**Comment by Si17br18** [comment]
Hi 

Thank you for the links, it would appear that integration was successful!

I am trying to do a test purchase, however, it is not letting me proceed saying card info is incorrect. Please see attached screen shot as well as test credentials. I have tried this with both my own card and the one provided and still receiving the same error

Look forward to hearing from you



Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18 the screenshot didn't come through? Can you try again or send to my email address (mgourd@vivawallet.com).

**Comment by matthewrgourd** [comment]
Hi @Si17br18 when using demo mode, you should only use the test card number provided. I know this seems a bit odd, but the failure may be because you didn't fill in a valid phone number. Can you putting in your real mobile number with no spaces? Thanks!

**Comment by Si17br18** [comment]
Hi 

Ive tried that solution but no luck
Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18 I think you're using the old version of our plugin, downloaded from here: 

https://developer.vivawallet.com/e-commerce-plugins/woocommerce/

You need to deactivate, delete, and install from the below page:

https://woocommerce.com/products/viva-wallet-for-woocommerce

**Comment by Si17br18** [comment]
Hi

I downloaded and installed the one you sent me this morning 
Viva Wallet for WooCommerce (v1.0.1)
Download page <https://woocommerce.com/products/viva-wallet-for-woocommerce>
Documentation <https://docs.woocommerce.com/document/viva-wallet-for-woocommerce>

Is this not the correct one?

Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Yes, that's correct! Sorry, let me check again and see what's going on here...

**Comment by matthewrgourd** [comment]
Hi @Si17br18 you've installed the plugin but I don't think you've copied your WooCommerce credentials from your Viva Wallet demo account. 
IN VIVA WALLET SELFCARE


1. [Log in](https://demo.vivapayments.com/en/signup) to your Viva Wallet demo account.
2. Go to **Settings > API access**.
3. Scroll down to _WooCommerce Credentials_ and click on **Generate a pair of credentials**.
Make a temporary copy of the values in notepad (or similar).

IN WORDPRESS

1. Go to **WooCommerce > Settings > Payments**.
2. Select the **Enable Viva Wallet** checkbox at the top of the page.
3. Tick the **Enable demo mode** checkbox.
4. Enter your **Demo Client ID** and **Demo Client Secret** (obtained from your Viva Wallet account earlier).<br />
A confirmation message will be displayed with the payment source code that the plugin set up automatically within your Viva Wallet account.

**Comment by Si17br18** [comment]
Hi

My credentials seem to be fine, please find Attached


Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18, sorry, I can't see the attachment. It appears you did follow the process but there's an error at the beginning of your Client ID. Please see below screenshot. The "http://" that I've indicated is not required. Where did you get this from out of interest? Can you also confirm that you're using the credentials from your demo Viva account and not live?

![Screenshot_2020-07-14_at_11_45_02](https://user-images.githubusercontent.com/44943019/87433940-cb755800-c5e1-11ea-8b92-d5242cc21f1b.png)

**Comment by Si17br18** [comment]
Hi 

My client Just copied and pasted them over so god knows lol!

 thank You ive made the Changes now but still same error. 

Demo Client: jyu1g1zgi0dc8a56r6ia89uo6284ax13spw1n1gbhej18.apps.vivapayments.com/ <http://jyu1g1zgi0dc8a56r6ia89uo6284ax13spw1n1gbhej18.apps.vivapayments.com/>
Demo Client Secret: vzeLGE0XHvn67225x7mJH9ngppPx2U
Live Client ID: aly3ef1ga6bxq6doq95lfqggxmcj0sznwvnkvs58t1f22.apps.vivapayments.com/ <http://aly3ef1ga6bxq6doq95lfqggxmcj0sznwvnkvs58t1f22.apps.vivapayments.com/>
Live Client Secret: sK9Z4N4KW22rhPUfXpi9tZ8j81wG3P


Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18 can you send me the Merchant ID for the client's demo account, please? I need to check the account details.

**Comment by Si17br18** [comment]
Hi 

Merchant  ID is : 19cb2f8e-685c-4490-a342-6497a04484c2
Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18 that's the Merchant ID for the live account so I'm going to obscure part of it immediately. Please send through the demo Merchant ID. If they don't have a demo account, simply get them to disable demo mode and the plugin will work. However, they'll have to test with a real card. But easy to do a refund from WooCommerce orders page.

**Comment by Si17br18** [comment]
Hi 

Just tried to do a normal payment and now getting the following error (see attached)


Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18 the source code hasn't been created so no payment will be possible. This should happen automatically so I think the credentials are still wrongly entered. Can you add me as a user to your WP site as I'm having difficulty working out what exactly the client is doing wrong! Email mgourd@vivawallet.com

**Comment by Si17br18** [comment]
Hi 

All set up for you
Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18 thanks for that. I can see that everything is correctly installed so it must be the credentials. Please double-check that they have been copied exactly as shown from the _WooCommerce Credentials_ section of the _API access_ page in the Viva Wallet banking app as indicated below. Unfortunately I can't check these for you as they're viewable only by the account owner.

![screencapture-members-vivawallet-selfcare-en-security-apiaccess-2020-07-14-16_52_18](https://user-images.githubusercontent.com/44943019/87447830-f0be9200-c5f2-11ea-8a80-820a9bbe6c41.png)

Thanks!

Matt

**Comment by Si17br18** [comment]
Hi 

Thank. You for looking into it. I managed to get access to my clients account and I copied the WC client ID as well as generated a new secret and still receiving the same issue
Kind regards,
Simon Brooker

Co-Founder
JCS Infusions Ltd

Tel: +44(0)790 413 3580
Whatsapp: +44(0)7377 391 441

**Comment by matthewrgourd** [comment]
Hi @Si17br18, I've checked and there was an errant forward slash at the end of the Client ID. I removed that and the payment source is now in the process of being verified. This can take up to 2 business days to complete. In the meantime, if the client wants to test using a demo account they should sign up for one [here](https://demo.vivapayments.com/en/signup). Payment source activation is immediate in this case. They can then switch to demo mode in the WC settings and test using the 4111 1111 1111 1111 card. Thanks!
